src/main/java/ch/uzh/ifi/hase/soprafs22/controller/LobbyController.java

```diff
@@ -81,5 +80,22 @@ public void updateLobby(@RequestHeader("token") String token, @PathVariable Long
 
         lobbyService.updateLobby(lobby, token, name, visibility, gameMode, gameType);
     }
+    @GetMapping("/v1/game/lobby")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public List<LobbyGetDTO> getLobbies() {
+
+        Collection<ILobby> lobbies = lobbyService.getLobbies();
+        List<LobbyGetDTO> lobbiesGetDTOs = new ArrayList();
+        Iterator var4 = lobbies.iterator();
+
+        while(var4.hasNext()) {
+            ILobby lobby = (ILobby) var4.next();
```

Comment:
remove the (ILobby). Downcasting is a code smell.

Reply:
I am not sure why it's downcasting since lobbies are also in type Ilobby - I followed the project's style of calling all objects Ilobby instead of lobby... is that not how we want to do it?

Reply:
Maybe because you transform the iterator<Ilobby> in a collection<Ilobby> but I am not sure.
If you move the iterator -> list code form the lobbyController to the lobby service and then return a list it is porbably easier. You could then do something like:

_List<ILobby<> lobbylist = new ArrayList<>();
lobbyManager.iterator().forEachRemaining(lobbylist::add);
return lobbylist;
_

Reply:
@apple00juice oh yes ok I agree with that, did not get this as a reference to downcasting. Will update, thanks!